src/tools/remote/src/main/java/com/google/devtools/build/remote/worker/OnDiskBlobStoreActionCache.java

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.remote.worker;
+
+import build.bazel.remote.execution.v2.Digest;
+import com.google.devtools.build.lib.remote.SimpleBlobStoreActionCache;
+import com.google.devtools.build.lib.remote.disk.OnDiskBlobStore;
+import com.google.devtools.build.lib.remote.options.RemoteOptions;
+import com.google.devtools.build.lib.remote.util.DigestUtil;
+import com.google.devtools.build.lib.vfs.Path;
+
+/** A {@link SimpleBlobStoreActionCache} backed by an {@link OnDiskBlobStore}. */
+class OnDiskBlobStoreActionCache extends SimpleBlobStoreActionCache {
+
+  public OnDiskBlobStoreActionCache(
+      RemoteOptions options,
+      Path cacheDir,
+      DigestUtil digestUtil) {
+    super(options, new OnDiskBlobStore(cacheDir), digestUtil);
+  }
+
+  public boolean containsKey(Digest digest) {
+    return ((OnDiskBlobStore) blobStore).contains(digest);
```

Comment:
Maybe add a `private final` field that's already casted instead of casting every time? 

I'm afraid casting every time will have runtime overhead before (if) JIT figures out that cast always succeeds, given how often `containsKey` will be called in large builds

Docs: [`checkcast`](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-6.html)

Reply:
Good point in general but this code is not performance critical as it's only used in tests.

Reply:
Still, it could be a good idea to 'not' make tests run longer ;)